eng/build.ps1

```diff
@@ -374,6 +377,14 @@ function TestUsingOptimizedRunner() {
       $dlls = Get-ChildItem -Recurse -Include "*.UnitTests.dll" -Exclude "*InteractiveHost*" $binDir
     }
   } elseif ($testVsi) {
+    # Capture the Environment variables altered by `InitializeDotNetCli` so we can revert to them
+    $alteredDotNetMultiLevelLookup = $env:DOTNET_MULTILEVEL_LOOKUP
+    $alteredPath = $env:PATH
+
+    # Resolve runtime, shared framework, or SDK from other locations for integration tests
```

Comment:
Should these instead be set inside of the code that launches devenv.exe in our test harness? My scenario is somebody who does something like:

1. Launch command line, restore and build. I think this sets the environment variable to add dotnet to the path.
2. devenv.exe Roslyn.sln to launch VS, which inherits those.
3. Run stuff from test explorer.

(i.e. the "how @jasonmalinowski develops" scenario.)

Although I'm not sure how in that case you get the "original" path. There's probably some Windows API to re-fetch the original environment block that the shell has before child processes inherited it.

Reply:
This would not be needed if we didn't call `InitializeDotNetCli` when running integration tests.

Reply:
@tmat I'll give it a shot. Since I wasn't as familiar with this part of the build, I was trying to preserve as much of the behavior as possible.

Reply:
hmm.. Not sure I did it right.